tests/conftest.py

```diff
@@ -70,10 +71,44 @@ def nodes_with_lower_outgoing_win_prob():
     """Nodes with outgoing ticket winning probability"""
     return ["6"]
 
-
 def random_distinct_pairs_from(values: list, count: int):
     return random.sample([(left, right) for left, right in itertools.product(values, repeat=2) if left != right], count)
 
+def first_two_from(values: list):
+    return [(left, right) for left, right in itertools.product(values, repeat=2) if left != right]
+
+@pytest.fixture(scope="function", autouse=True)
+def config_to_yaml(request):
+    """
+    Fixture to convert the nodes_config parameter to yaml
+    """
+    if "nodes_config" in request.fixturenames:
+        nodes_config = request.getfixturevalue("nodes_config")
+        yaml_config = yaml.dump(nodes_config, default_flow_style=False)
+        request.config.cache.set("yaml_config", yaml_config)
+
+@pytest.fixture(scope="function")
+async def swarm3(request):
+    params_path = PWD.joinpath("sdk/python/localcluster.params.yml")
+
+    # load cache created by autouse fixture config_to_yaml
+    extra_params = request.config.cache.get("yaml_config", None)
+    if extra_params is not None:
+        extra_params = yaml.safe_load(extra_params)
+        request.config.cache.set("yaml_config", None)
+
+    cluster, anvil = await localcluster.bringup(params_path, test_mode=True, fully_connected=False, use_nat=True, extra_env=extra_params)
+
+    yield cluster.nodes
+
+    cluster.clean_up()
+    anvil.kill()
```

Comment:
Please have a look at the https://github.com/hoprnet/hoprnet/pull/7131 and whether it is relevant to you.